<a href="https://colab.research.google.com/github/RujanBhuju/Discretemath/blob/main/Project_4.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Project Part 2 : CPSMA 3913 Dr. Jacob's Discrete Mathematics

Counting

Rujan Bhuju : 268963

According to the condition given for simulation I will be using a punch code lock which requires 3 digit to unlock it.The numbers we can choose from are from 0-5.

Thus, (n) is the number of digits to choose from & (r) is the number of digit we choose.



To find The number of outcomes possible if it is considered a sequence in that particular order.
 $$= n^r$$


In [9]:
import math
6**3

216

Thus , the required number of outcomes as per the given condition is 216.

To find the number of outcomes possible in my simulation if order did not matter and also we can repeate the numbers.

$$= (r+n-1)/((n-r)!*r!)$$


In [8]:
math.factorial(3+6-1) / ((math.factorial(6 - 3)) * math.factorial(3))

1120.0

Thus , the required number of outcomes as per the given condition is 1120.

To find the number of outcomes that is possible if we were not allowed to select an item multiple times with and without the importance of order.

1) when not allowed to select multiple times with order.

$$= n! / (n-r)!$$

In [11]:
math.factorial(6) / (math.factorial(6 - 3)) 

120.0

Thus , the required number of outcomes as per the given condition is 120.

2) when not allowed to select multiple times without order.

$$= n!/((n-r)!*r!)$$

In [7]:
math.factorial(6) / ((math.factorial(6 - 3)) * math.factorial(3))

20.0

Thus , the required number of outcomes as per the given condition is 20.

To find the coefficent of the 5th order of (x − 3)9.

We will be using binomial theorem

$$
(a+b)^n = \sum_{k=0}^n {n\choose{k}} a^{n-k} b^k
$$

Using our expression (x − 3)9.

$$
(x-3)^9 = \binom{9}{0} x^9(-3)^0 + \binom{8}{1} x^8(-3)^1 + \binom{7}{2} x^7(-3)^2 + \binom{6}{3} x^6(-3)^3 + \binom{5}{4} x^5(-3)^4 + \binom{4}{5} x^4(-3)^5 + \binom{3}{6} x^3(-3)^6 + \binom{2}{7} x^2(-3)^7 + \binom{1}{8} x^1(-3)^8 + \binom{0}{9} x^0(-3)^9
$$

Solving the equation

we get,

$$
\binom{5}{0} x^5(-3)^0 + \binom{4}{1} x^4(-3)^1 + \binom{3}{2} x^3(-3)^2 + \binom{2}{3} x^2(-3)^3 + \binom{1}{4} x^1(-3)^4 + \binom{0}{5} x^0(-3)^5 \\ = \ x^5 - 12x^4 + 27x^3 
$$

Since the coefficient of the fifth order expression $$x^5$$ is 1,

The coefficent of the 5th order of (x − 3)9 is 1.
